test/test_ops.py

```diff
@@ -172,6 +189,33 @@ def test_numpy_ref(self, device, dtype, op):
         finally:
             torch.set_default_dtype(cur_default)
 
+    # Same as above `test_numpy_ref`, but with different settings required for the MPS backend
+    # When MPS becomes more consistent, this can probably be merged with the above using
+    # `@dtypesIfMPS(torch.float32)`, but for now, the assertions themselves need to be loosened
+    @unittest.skipIf(TEST_WITH_ASAN, "Skipped under ASAN")
+    @onlyMPS
+    @suppress_warnings
+    # MPS only supports float32
+    @ops(_ref_test_ops, allowed_dtypes=(torch.float32,))
```

Comment:
Should this be `float64`? In the test the default dtype is set to `double` to be more consistent with NumPy, but the test is currently run in `float32`.

Reply:
If I do that, literally everything fails with
`TypeError: Cannot convert a MPS Tensor to float64 dtype as the MPS framework doesn't support float64. Please use float32 instead.`
so I'm assuming this is correct and it's a limitation of MPS? It's a real pain, too! The `broken_on_ref_inputs` are mostly failing on tests that test type promotion to float64

Reply:
That's understandable, yep. In that case I would also remove the setting of the default dtype within the test for consistency, and add a comment that, at the time the test was added, MPS didn't support double, which is why the test compares with NumPy in float32

Reply:
Done. This broke a few more tests which seem to mistakenly promote to float64, so I skipped those as well.